src/package-linker.js

```diff
@@ -380,6 +387,10 @@ export default class PackageLinker {
         linkBinConcurrency,
       );
     }
+
+    for (const [, {pkg}] of flatTree) {
+      await this._warnForMissingBundledDependencies(pkg);
```

Comment:
I made a separate `_warnForMissingBundledDependencies` function primarily as a Single-Responsibility Principle / Separation of concerns nod. `linkSelfDependencies` does not sound like a function that would check bundled dependencies.

If I thought it would make a big performance difference I probably would have combined them, but I can't imagine that many packages actually have bundled deps, so I'm figuring the vast majority of installs won't see any noticeable performance difference.

I'm open to refactoring it though, if you want.